hedera-node/hapi-utils/src/main/java/com/hedera/node/app/hapi/utils/exports/FileSignTool.java

```diff
@@ -314,7 +312,8 @@ private static void createSignatureFileForRecordFile(
                 }
             }
         }
-        LOGGER.info(MARKER, "Record stream file header is {}", Arrays.toString(fileHeader));
+        final var fileHeaderString = Arrays.toString(fileHeader);
```

Comment:
We should check the log level before doing this or do this in a lambda that is passed to the logger.

Reply:
Fixed.
Out of curiosity: why do we need to do that exactly here? Is `Arrays.toString(fileHeader);` an expensive operation?

Reply:
I am always cautious when Strings, arrays, or collections are involved. Also, I would always try to avoid any work for info level, because you never use that in production.